packages/aragon-wrapper/src/index.js

```diff
@@ -1242,6 +1246,99 @@ export default class Aragon {
     return []
   }
 
+  /**
+   * Calculate the transaction path for a basket of intents.
+   * Expects the `intentBasket` to be an array of tuples holding the following:
+   *   {string}   destination: destination address
+   *   {string}   methodName: method to invoke on destination
+   *   {Array<*>} params: method params
+   * These are the same parameters as the ones used for `getTransactionPath()`
+   *
+   * Allows user to specify how many of the intents should be checked.
+   * `checkMode` supports:
+   *   'all': All intents will be checked to make sure they use the same forwarding path.
+   *   'single': assumes all intents can use the path found from the first intent
+   *
+   * @param  {Array<Array<string, string, Array<*>>>} intentBasket Intents
+   * @param  {Object} [options]
+   * @param  {string} [options.checkMode] Path checking mode
+   * @return {Promise<Array<Object>>} An array of Ethereum transactions that describe each step in the path
+   */
+  async getTransactionPathForIntentBasket (intentBasket, { checkMode = 'all' } = {}) {
+    // Get transaction paths for entire basket
+    const intentsToCheck =
+      checkMode === 'all'
+        ? intentBasket // all -- use all intents
+        : checkMode === 'single'
+          ? [intentBasket[0]] // single -- only use first intent
+          : []
+    const intentPaths = await Promise.all(
+      intentsToCheck.map(
+        ([destination, methodName, params]) =>
+          this.getTransactionPath(destination, methodName, params))
+    )
+
+    // Check paths all match
+    const individualPaths = intentPaths
+      // Map each path to just be an array of destination addresses
+      .map(path =>
+        path.map(({ to }) => to)
+      )
+      // Take each array of destination addresses and create a single string
+      .map(path => path.join('.'))
+    // Check if they all match by seeing if a unique set of the individual path
+    // strings is a single path
+    const pathsMatch = (new Set(individualPaths)).size === 1
+
+    if (!pathsMatch) {
+      return []
+    }
+
+    // Let's create direction transactions for each intent in the intentBasket
+    const sender = (await this.getAccounts())[0] // TODO: don't assume it's the first account
+    const directTransactions = await Promise.all(
+      intentBasket.map(
+        async ([destination, methodName, params]) =>
+          createDirectTransaction(sender, await this.getApp(destination), methodName, params, this.web3)
+      )
+    )
+
+    if (intentPaths[0].length === 1) {
+      // Sender has direct access
+      return {
+        direct: true,
+        transactions: directTransactions
```

Comment:
Don't we need to `applyTransactionGas` to these transactions before returning them?

Reply:
Hmm yes... we should also apply gas to https://github.com/aragon/aragon.js/blob/ca9f562a4b14783ced5087e2aae8217453e2e8cf/packages/aragon-wrapper/src/index.js#L1776.

In the case where apply the gas fails, I guess we should just return an empty array to signal that the transaction wouldn't succeed.